### ISSUE TITLE
Not compatible with MySql Assembly

### ISSUE THREAD

**Issue by FireInWinter** [PowerShell/PowerShell#4666]
I saw this in an internal module using MySql and also using Invoke-MySqlQuery from the PowerShell Gallery.  I get an error when I try to use it on Beta 5 on Windows 10 or Windows 8.1.

My understanding of the .Net Standard was that standard .Net assemblies should work on Windows in PowerShell Core 6.  When I try it with the MySql Connector/Net 6.9.9 it doesn't.  I've tried it on a different system using Connector/Net 8.0.8 and had the same symptom.  In either case it gets an error when I try to do a Conn.Open.  I've simplified the steps to reproduce to get an error in either case, but the actual issue is to connect, but it isn't getting far enough to even try.

I don't expect this to work on the Linux/MaxOS version of PowerShell Core, but I would expect it to be able to work on Windows.

Steps to reproduce
------------------

Install MySql Connector/Net 6.9.9 
In PowerShell Core 6 Beta 5:
```
Add-Type -AssemblyName 'MySql.Data, Version=6.9.9.0, Culture=neutral, PublicKeyToken=c5687fc88969c44d'
$Conn = New-Object MySql.Data.MySqlClient.MySqlConnection
$Conn.ConnectionString = 'server=BogusServer1'
$Conn.Open()
```

Expected behavior
-----------------
This should take a few seconds to generate the error.
```
Exception calling "Open" with "0" argument(s): "Unable to connect to any of the specified MySQL hosts."
```

Actual behavior
---------------
This returns an error immediately.
```
Exception calling "Open" with "0" argument(s): "The type initializer for 'MySql.Data.MySqlClient.Replication.ReplicationManager' threw an exception."
```

Environment data
----------------

OS: Windows 8.1 or 10

```
$PSVersionTable

Name                           Value                                                                                                                                                                                       
----                           -----                                                                                                                                                                                       
PSVersion                      6.0.0-beta                                                                                                                                                                                  
PSEdition                      Core                                                                                                                                                                                        
GitCommitId                    v6.0.0-beta.5                                                                                                                                                                               
OS                             Microsoft Windows 6.3.9600                                                                                                                                                                  
Platform                       Win32NT                                                                                                                                                                                     
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                     
PSRemotingProtocolVersion      2.3                                                                                                                                                                                         
SerializationVersion           1.1.0.1                                                                                                                                                                                     
WSManStackVersion              3.0                                                                                                                                                                                         
```

**Comment by FireInWinter** [comment]
Looks like a related issue happens if trying to see the exported types from an assembly.

```
[appdomain]::CurrentDomain.GetAssemblies() | ? fullname -match 'MySql.Data' | % { $_.GetExportedTypes() }
```
Genererates the error:
```
Exception calling "GetExportedTypes" with "0" argument(s): "Could not load file or assembly 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified."
```
On Windows PowerShell it generates a list of exported types.

**Comment by SteveL-MSFT** [comment]
Can you try installing https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.Configuration.ConfigurationManager/4.4.0-preview2-25405-01

Just follow the PowerShellGet instructions

**Comment by kensel** [comment]
@SteveL-MSFT I have the same error using the latest docker powershell on linux and installing the SimplySQL module.  running the command above gives the following even after adding the package and adding the type manually.

```
PS /> $ObjMysql = New-Object MySql.Data.MySqlClient.MySqlConnection($ConnStr)
PS /> $ObjMysql.GetType()                                                                                                 [84/1969]

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------                                                                True     False    MySqlConnection                          System.Data.Common.DbConnection                                                                                                                                                                            
PS /> $ObjMysql.Open()                                                                                                             Exception calling "Open" with "0" argument(s): "The type initializer for 'MySql.Data.MySqlClient.Replication.ReplicationManager' th
rew an exception."
At line:1 char:1                                                                                                                   + $ObjMysql.Open()                                                                                                                 + ~~~~~~~~~~~~~~~~                                                                                                                 + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
+ FullyQualifiedErrorId : TypeInitializationException

PS /> Add-Type -Path /usr/local/share/PackageManagement/NuGet/Packages/System.Configuration.ConfigurationManager.4.4.1/lib/netstandard2.0/System.Configuration.ConfigurationManager.dll
PS /> [appdomain]::CurrentDomain.GetAssemblies() | ? fullname -match 'MySql.Data' | % { $_.GetExportedTypes() }                    Exception calling "GetExportedTypes" with "0" argument(s): "Could not load file or assembly 'System.Configuration.Install, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
"
At line:1 char:83
+ ... lies() | ? fullname -match 'MySql.Data' | % { $_.GetExportedTypes() }
+                                                   ~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
+ FullyQualifiedErrorId : FileNotFoundException

```

**Comment by cabozone** [comment]
I have this working with Debian 9 + PowerShell Core 6.1.2 + Mono-Project + MySQL .NET/Mono

https://youtu.be/DMhTtrM5v4w

**Comment by fak0r** [comment]
For the record:
Yes, Debian/Ubuntu + PS Core + Mono + Connector/Mono works but.....
I think the performance is very bad.
My script runtimes have nearly tripled compared to Windows + PS Core + Connector/NET
I guessed that Connector/Mono is the problem.
Therefore I rewrote my scripts to PHP/PDO for testing purposes. There I have the same performance as under Windows.

If speed is important, PHP/PDO might be an easy alternative
